pkg/workos_errors/authentication_errors.go

```diff
@@ -0,0 +1,204 @@
+package workos_errors
+
+import (
+	"errors"
+)
+
+// Authentication error code constants
+const (
+	EmailVerificationRequiredCode                 = "email_verification_required"
+	MFAEnrollmentCode                             = "mfa_enrollment"
+	MFAChallengeCode                              = "mfa_challenge"
+	OrganizationSelectionRequiredCode             = "organization_selection_required"
+	SSORequiredCode                               = "sso_required"
+	OrganizationAuthenticationMethodsRequiredCode = "organization_authentication_methods_required"
+)
+
+// AuthenticationFactor represents an MFA factor
+type AuthenticationFactor struct {
+	ID   string `json:"id"`
+	Type string `json:"type"`
+}
+
+// Organization represents an organization in selection error
+type Organization struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+// User represents a user in authentication errors
+type User struct {
+	Object            string `json:"object"`
+	ID                string `json:"id"`
+	Email             string `json:"email"`
+	FirstName         string `json:"first_name"`
+	LastName          string `json:"last_name"`
+	EmailVerified     bool   `json:"email_verified"`
+	ProfilePictureURL string `json:"profile_picture_url"`
+	CreatedAt         string `json:"created_at"`
+	UpdatedAt         string `json:"updated_at"`
```

Comment:
I think the docs may be slight off here. I believe this returns a full `User` object. Should we just reused the existing `User` struct from UserManagement?